perf_benchmark/benchmark_common.cc

```diff
@@ -0,0 +1,294 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+//
+#include <fstream>
+#include <sstream>
+#include <utility>
+#include "perf_benchmark/benchmark_common.h"
+#include "google/protobuf/text_format.h"
+#include "absl/strings/escaping.h"
+#include "nlohmann/json.hpp"
+
+namespace google {
+namespace grpc {
+namespace transcoding {
+
+namespace perf_benchmark {
+
+namespace pb = ::google::protobuf;
+
+std::string LoadFile(const std::string& file_name) {
+  std::ifstream ifs(file_name);
+  if (!ifs) {
+    std::cerr << "Could not open " << file_name.c_str() << std::endl;
+    return {};
+  }
+  std::ostringstream ss;
+  ss << ifs.rdbuf();
+  return ss.str();
+}
+
+bool LoadService(const std::string& config_pb_txt_file,
```

Comment:
Instead of returning bools for error handling, return absl::Status here (or absl::StatusOr<std::string> for LoadFile)

https://abseil.io/docs/cpp/guides/status

Reply:
You can also use these macros for improved readability

https://github.com/protocolbuffers/protobuf/blob/a3c8e2deb05186334b7ee8c1174f44802e38b43d/src/google/protobuf/stubs/status_macros.h#L53

```
RETURN_IF_ERROR
ASSIGN_OR_RETURN
```